impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -38,65 +35,61 @@
-            xml.setAddDefaultEntities(request.isAddDefaultEntities());
-            if (inputStream != null) {
-                return xml.read(inputStream, request.isStrict());
-            } else if (reader != null) {
-                return xml.read(reader, request.isStrict());
-            } else if (path != null) {
-                try (InputStream is = Files.newInputStream(path)) {
-                    return xml.read(is, request.isStrict());
+            if (request.getInputStream() != null) {
+                return reader.read(request.getInputStream(), request.isStrict());
+            } else if (request.getReader() != null) {
+                return reader.read(request.getReader(), request.isStrict());
+            } else if (request.getPath() != null) {
+                try (var is = Files.newInputStream(request.getPath())) {
+                    return reader.read(is, request.isStrict());
                 }
             } else {
-                try (InputStream is = url.openStream()) {
-                    return xml.read(is, request.isStrict());
+                try (var is = request.getURL().openStream()) {
+                    return reader.read(is, request.isStrict());
                 }
             }
-        } catch (Exception e) {
+        } catch (IOException | XMLStreamException e) {
             throw new XmlReaderException("Unable to read plugin: " + getMessage(e), getLocation(e), e);
         }
     }
 
-    @Override
-    public void write(XmlWriterRequest<PluginDescriptor> request) throws XmlWriterException {
-        nonNull(request, "request");
-        PluginDescriptor content = nonNull(request.getContent(), "content");
-        Path path = request.getPath();
-        OutputStream outputStream = request.getOutputStream();
-        Writer writer = request.getWriter();
-        if (writer == null && outputStream == null && path == null) {
-            throw new IllegalArgumentException("writer, outputStream or path must be non null");
-        }
+    private static PluginDescriptorStaxReader setAddDefaultEntities(
+            XmlReaderRequest request, PluginDescriptorStaxReader reader) {
+        reader.setAddDefaultEntities(request.isAddDefaultEntities());
+        return reader;
+    }
+
+    private static void write(XmlWriterRequest<PluginDescriptor> request, PluginDescriptorStaxWriter writer) {
         try {
-            if (writer != null) {
-                new PluginDescriptorStaxWriter().write(writer, content);
-            } else if (outputStream != null) {
-                new PluginDescriptorStaxWriter().write(outputStream, content);
+            if (request.getWriter() != null) {
+                writer.write(request.getWriter(), request.getContent());
+            } else if (request.getOutputStream() != null) {
+                writer.write(request.getOutputStream(), request.getContent());
             } else {
-                try (OutputStream os = Files.newOutputStream(path)) {
```

Comment:
kind of dry: `OutputStream`. Context could not be shorter. @elharo This is good example why its good pattern in terms of functional programming, avoiding overhead.

<img width="442" alt="image" src="https://github.com/user-attachments/assets/35c257ad-c1f2-434b-8f84-1c1e6c896252" />

Reply:
I prefer the original. var doesn't help

Reply:
yes, its irrelevant detail. Suggested smallest and final version. This could be redone in future. Lets dram about `val` cause actually this is.

Reply:
done.